Suspension API v1 Spec (#66)
BinghamtonRover/RoverSystem: docs/suspension.md

# Suspension API, Version 1

This document describes the rover's suspension command API of the version indicated above.

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.

## Purpose

This API defines the interface with the 2018-2019 IEEE Mars Rover Suspension system. 

This document is broken down into several components:

- Purpose
- General Functionality
	- Operations
	- Safety Mechanisms
  - Command Hierarchy
- Commands

## General Functionality

The API allows for manual control of the suspension system.

### Operations

We define the operations and their effects for both automated and manual control modes. 

#### Manual Control

> The speed and direction of the left and right-side drive motors can be manaully selected.

Reviewer: ```suggestion
The speed and direction of the left and right-side drive motors can be manually selected.
```

> The speed and direction of the left and right-side drive motors can be manaully selected.

Reviewer: ```suggestion
The speed and direction of the left and right side drive motors can be manually selected.
```

### Safety Mechanisms

> In order for the suspension system to continue movement, it requires constant reasurance. If a signal is not recieved from the computer within (5) seconds of the last, the drive motors will be immediately halted.

Reviewer: ```suggestion
In order for the suspension system to continue movement, it requires constant reassurance. If a signal is not received from the computer within (5) seconds of the last, the drive motors shall be immediately halted.
```

### Command Hierarchy

> Command precidence is based on the time-of-execution, with the most-recent command taking precidence.

Reviewer: ```suggestion
Command precedence is based on the time-of-execution, with the most-recent command taking precedence.
```
